File: Week04/LeetCode-0219.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        /// O(N) algorithm with O(N) extra space
        ///          <num, index>
        unordered_map<int, int> myMap;
        for (int i=0; i<nums.size(); i++){
            if (myMap.count(nums[i]) != 0)
                if (i-myMap[nums[i]] <= k)
                    return true;
            myMap[nums[i]] = i;
        } // end-for
        
        return false;
    } // end-containsNearbyDuplicate
};

File: Week04/LeetCode-0219.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        """
        O(N) algorithm with O(N) extra space
        """
        myMap = dict()
        for i in range(len(nums)):
            if nums[i] in myMap:                
                if i-myMap[nums[i]] <= k:
                    return True
            myMap[nums[i]] = i        
        return False
        